test/test_tensordict.py

```diff
@@ -3714,6 +3715,187 @@ def test_tensordict_prealloc_nested():
     assert buffer["agent.obs"].batch_size == torch.Size([B, N, T])
 
 
+def test_tensordict_view_iteration():
```

Comment:
Some of this can likely be unified with [`test_keys_view`](https://github.com/pytorch-labs/tensordict/blob/3f20f804dc36e58c8fc20f69ceab526f49d5df86/test/test_tensordict.py#L2963)

Reply:
We should actually create a full class for all tests regarding TensorDictKeysView same way as for TensorDict tests